src/test_runner.cpp

```diff
@@ -3,19 +3,57 @@
 #include <iostream>
 #include <string>
 
+#include "ast/element.h"
+#include "evaluator/error.h"
+#include "evaluator/evaluator.h"
+#include "evaluator/expression.h"
 #include "reader/error.h"
 #include "reader/reader.h"
 
+using evaluator::EvaluationError;
+using evaluator::Evaluator;
+using evaluator::Program;
 using reader::Reader;
 
-bool test_fail_file(std::filesystem::path path) {
+enum class Mode { SYNTAX, SEMANTIC };
+
+enum class ArgumentErrorCause { ExtraArgument, WrongFlagPosition };
+
+class ArgumentError : public std::exception {
+    ArgumentErrorCause cause;
+    std::string_view argument;
+
+  public:
+    ArgumentError(ArgumentErrorCause cause, std::string_view argument)
+        : cause(cause), argument(argument) {}
+
+    friend std::ostream&
+    operator<<(std::ostream& stream, ArgumentError const& error) {
+        stream << "Error: ";
+
+        if (error.cause == ArgumentErrorCause::ExtraArgument) {
+            stream << "extra argument: '" << error.argument << "'";
+        }
+
+        else if (error.cause == ArgumentErrorCause::WrongFlagPosition) {
+            stream << "wrong flag position: '" << error.argument << "'";
+        }
+
+        return stream;
+    }
+};
+
+bool test_fail_file_syntax(std::filesystem::path path) {
     std::ifstream file(path);
     std::string line;
 
     while (std::getline(file, line)) {
         try {
             Reader reader((std::string_view(line)));
             auto elements = reader.read();
+
+            std::cout << "This file is supposed to be syntactically incorrect, "
+                         "but it is actucally not.";
```

Comment:
it is not syntactially incorrect --- сложно читается

```suggestion
            std::cout << "This file is supposed to be syntactically incorrect, "
                         "but it parsed without errors";
```